src/Raven.Client/Json/BlittableOperation.cs

```diff
@@ -159,6 +161,37 @@ public static bool EntityChanged(BlittableJsonReaderObject newObj, DocumentInfo
             return true;
         }
 
+        private static unsafe bool CompareStringsWithEscapePositions(JsonOperationContext context, BlittableJsonReaderObject.PropertyDetails oldProp,
+            BlittableJsonReaderObject.PropertyDetails newProp)
+        {
+            // this is called if the values are NOT equal, but we need to check if the oldProp was read from network and already resolved 
+            // the escape characters
+
+            if (oldProp.Value is LazyStringValue lsv)
+            {
+                int pos = lsv.Size;
+                int numOfEscapePositions = BlittableJsonReaderBase.ReadVariableSizeInt(lsv.Buffer, ref pos);
+                if (numOfEscapePositions == 0)
+                    return false;
+
+                var memoryStream = new MemoryStream();
+                using (var textWriter = new AsyncBlittableJsonTextWriter(context, memoryStream, CancellationToken.None))
```

Comment:
why Async? there is a `BlittableJsonTextWriter`

Reply:
Done